spec/base_model_spec.rb

```diff
@@ -5,7 +5,8 @@
       "name"            => "jeff",
       "address"         => { "street" => "22 charlotte rd"},
       "serialized_json" => { "key" => "value"}.to_json,
-      "serialized_yaml" => { "key" => "value"}.to_yaml
+      "serialized_yaml" => { "key" => "value"}.to_yaml,
+      ":Special$ Key3"  => { "key" => "value"}
```

Comment:
According to the BZ, he was creating an ec2 instance with a /dev/xvda in it. When `/dev/xvda` being the key, the parsing and conversion code fails. I could change the test to use `/dev/xvda`.

Reply:
It sounds like this is buried in extra vars or something that we probably shouldn't be parsing and turning into methods, right?

Reply:
I am attempting to give a general solution to avoid coming back when another similar case arises.
Consider the following json
```
{
  "abc" : {
    "xyz": "value1",
    "/dev/xvda": {
      "mykey": "myvalue"
    }
  }
}
```
Ideally we should want to access `value1` through `abc.xyz`. The second key can be accessed either by `abc._dev_xvda` or `abc['/dev/xvda']`, but its value will still be a converted model rather than raw hash. 

The problem reported in BZ is not an extra var. Those we should not parse and convert are already wrapped in string (not parsable).

Reply:
I guess what I'm trying to say is that the code is currently trying to parse everything and turn it into classes and methods and we should probably only do that for the known classes.

Reply:
I am leaning with @bdunne with this.  `/dev/xda` caught us off guard here, which means more odd cases will show up later.  I think we should be white listing classes and only constantize name spaced `AnsibleTowerClient` classes a lot like how `safe_constantize` works.  Depending on a regex feels brittle.

Reply:
The purpose of auto conversion in base model is to provide convenience for method style access to (nested) properties. It does NOT fundamentally change the data. You can always access the raw data treating it as a hash, ignoring any auto generated methods and classes.

Secondly the name of the auto-generated embedded class does not matter. It can be anonymous or any random valid name because what we really want is the methods. Here we just try to make the class name more meaningful.

If we want to white list classes, then we should white list methods too. Since we need to white list everything we should really abandon the base class, at least the auto conversion part. In stead we should create every class with known attributes. I am fine with this approach, but it will not to be addressed by this PR.